code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -286,10 +286,16 @@
 
 /obj/item/reagent_containers/glass/bucket/equipped(mob/user, slot)
 	..()
-	if(slot == SLOT_HEAD && reagents.total_volume)
-		to_chat(user, "<span class='userdanger'>[src]'s contents spill all over you!</span>")
-		reagents.reaction(user, TOUCH)
-		reagents.clear_reagents()
+	if (slot == SLOT_HEAD)
+		if(reagents.total_volume)
+			to_chat(user, "<span class='userdanger'>[src]'s contents spill all over you!</span>")
+			reagents.reaction(user, TOUCH)
+			reagents.clear_reagents()
+		container_type = NONE
+
+/obj/item/reagent_containers/glass/bucket/dropped(mob/user)
+	..()

```

Comment:
I'm having a brain-toot, if `..()` `return`s then the container **will** or **will not** still be set back?

also it might be wise to have `container_type = initial(container_type)`

Reply:
You could just put this line below the container_type or `. = ..()`

Reply:
Doing .=..() at the top would probably be best so it drops and then sets the container type, I think.